### PR TITLE
Improve Mapping documentation.

### PR DESCRIPTION
**Pull request by bangerth**
Add some documentation. Put functions and classes in grouped
environments.

### CODE REVIEW

File: include/deal.II/fe/mapping.h [hunk 1]
```diff
@@ -233,27 +233,52 @@ class Mapping : public Subscriptor
   virtual ~Mapping ();
 
   /**
+   * Return a pointer to a copy of the present object. The caller of this copy
+   * then assumes ownership of it.
+   *
+   * Since one can't create objects of class Mapping, this function of course
+   * has to be implemented by derived classes.
```
Comment (tamiko) [inline include/deal.II/fe/mapping.h]: maybe "is marked as virtual and has to be implemented..."?

File: include/deal.II/fe/mapping.h [hunk 2]
```diff
@@ -291,6 +324,61 @@ class Mapping : public Subscriptor
     const Point<spacedim> &p) const;
 
   /**
+   * @}
+   */
+
+
+  /**
+   * Returns whether the mapping preserves vertex locations, i.e. whether the
```
Comment (tamiko) [inline include/deal.II/fe/mapping.h]: ..., i.e., ...

File: include/deal.II/fe/mapping.h [hunk 3]
```diff
@@ -291,6 +324,61 @@ class Mapping : public Subscriptor
...
+
+
+  /**
+   * Computing the mapping between a real space point and a point in reference
+   * space failed, typically because the given point lies outside the cell
+   * where the inverse mapping is not unique.
+   *
+   * @ingroup Exceptions
+   */
+  DeclException0(ExcTransformationFailed);
```
Comment (tamiko) [inline include/deal.II/fe/mapping.h]: What about we use this documentation update to also update the exception to ```DeclExceptionMSG``? This is also a documentation update in some sense :-]

File: include/deal.II/fe/mapping.h [hunk 4]
```diff
@@ -823,6 +723,156 @@ class Mapping : public Subscriptor
...
+  void
+  transform (const VectorSlice<const std::vector< DerivativeForm<1, dim, spacedim> > > input,
+             VectorSlice<std::vector<Tensor<2,spacedim> > >             output,
+             const InternalDataBase &internal,
+             const MappingType type) const = 0;
+
+
+  /**
+   * Transform a tensor field from the reference cell to the physical cell.
+   * This tensors are most of times the jacobians in the reference cell of
```
Comment (tamiko) [inline include/deal.II/fe/mapping.h]: "most of the times"?

File: include/deal.II/fe/mapping.h [hunk 5]
```diff
@@ -823,6 +723,156 @@ class Mapping : public Subscriptor
...
+   * <li> @p mapping_piola: A field of <i>n-1</i>-forms on the reference cell
+   * is also represented by a vector field, but again transforms differently,
+   * namely by the Piola transform
+   * @f[
+   *  \mathbf u(\mathbf x) = \frac{1}{\text{det}J(\mathbf x)}
+   * J(\mathbf x) \mathbf{\hat u}(\mathbf x).
+   * @f]
+   * </ul>
+   *
+   * @todo What is n in mapping_piola description?
```
Comment (tamiko) [inline include/deal.II/fe/mapping.h]: Can you remove the ```@todo``` and add a line that the ```n``` stands for the space dimension? (Applies also to other places.)

The formulas definitely deserve some checking. But I'm to tired right now to read pure latex math statements. :-]

**Comment by bangerth** [comment]
@tamiko : I've updated all of the locations you marked. They were all pre-existing, as I only moved stuff around (with a couple actual updates). Someone should really go through these classes and check all of the formulas, but that's not going to be me.

**Pull request merged**